### ISSUE TITLE
Kernel.spawn doesn't support some options (ex: :chdir)

### ISSUE THREAD

**Issue by tagomoris** [jruby/jruby#985]
Some options like :pgroup works, but others (ex: :chdir, filehandles) doesn't work.

```
$ jruby -e 'pid = spawn("sleep 5", :pgroup => 0, :chdir => "/", STDOUT => :close); p pid; Process.detach(pid)'
file:/Users/tagomoris/.rbenv/versions/jruby-1.7.4/lib/jruby.jar!/jruby/kernel19/process.rb:4 warning: unsupported spawn option: chdir
TypeError: wrong exec option: RubyIO(66, 1)
  _spawn_internal at org/jruby/RubyProcess.java:1067
            spawn at file:/Users/tagomoris/.rbenv/versions/jruby-1.7.4/lib/jruby.jar!/jruby/kernel19/process.rb:3
            spawn at file:/Users/tagomoris/.rbenv/versions/jruby-1.7.4/lib/jruby.jar!/jruby/kernel19/kernel.rb:29
           (root) at -e:1
```

**Comment by jcode** [comment]
any update on this? We're having problems using the schmooze gem :(

**Comment by doudou** [comment]
chdir seems to work fine on 9.1.7.0 with `system` and `spawn` ... but not `exec`

**Comment by headius** [comment]
chdir for system and spawn is achieved by modifying the command line to run as an sh script that chdir's first. Unfortunately a similar approach with popen breaks other things, so we can't support that yet.

**Comment by headius** [comment]
The original example appears to run correctly.

The example provided by @BanzaiMan still fails to execute:

```
$ rm -f bar; ruby -v -e 'exec("date", :chdir => "/", STDOUT => "bar")'; cat bar
jruby 9.3.0.0-SNAPSHOT (2.6.5) 2021-06-15 36b083365a OpenJDK 64-Bit Server VM 11.0.4+11 on 11.0.4+11 +jit [darwin-x86_64]
uri:classloader:/jruby/kernel/kernel.rb:4: warning: unsupported exec option: chdir
Tue Jun 15 16:20:00 CDT 2021
cat: bar: No such file or directory
```

The `Open3.popen2e` example should work correctly but the example is not a free-standing example. There has been work in the past couple years to improve the compatibility of open3 and it now nearly the same code as CRuby.

There are other issues filed for more specific examples not mentioned here, so I will close this one as fixed in 9.3.
